app/src/main/java/org/simple/clinic/sync/SyncConfigModule.kt

```diff
@@ -29,6 +30,27 @@ class SyncConfigModule {
     )
   }
 
+  /**
+   * Drug sync doesn't support batch size at API level yet. So using the normal batch size can cause
+   * API loop if the payloads size is greater than the batch size. So, to avoid that issue and also
+   * support batching in case server implements batching support for this API in future, we will have
+   * a separate `SyncConfig` for the `DrugSync`.
+   * 
+   * We can remove this once medications API has added the batching support and replace this usage
```

Comment:
Can we create tickets in Clubhouse to track this instead of leaving it as a comment here? Also, is this expected to be done before this feature goes to `PROD`? I'm concerned about leaving a potential infinite sync loop case in `PROD`,

Reply:
The reason is that the remote config sync currently happens _after_ all the other syncs are done. So in the event that we do run into this case in `PROD`, we might end up in a situation where bumping the batch size remotely has no effect in `PROD` because the remote config does not update because it is waiting for the drug sync to complete.

Reply:
It's not, according to Prabhanshu we are going to add the batching support if the medications cross 1000 or something like that.

https://simpledotorg.slack.com/archives/C01P131R1GR/p1626094192001300


>The reason is that the remote config sync currently happens after all the other syncs are done. So in the event that we do run into this case in PROD, we might end up in a situation where bumping the batch size remotely has no effect in PROD because the remote config does not update because it is waiting for the drug sync to complete.

That can be an issue, currently, we want to make sure the older versions also has support for batching. So when we eventually end up adding batching support we should first update the remote config and also the app (to go back to normal sync config) and then roll the server changes to `PROD`. wdyt?

Reply:
> Can we create tickets in Clubhouse to track this instead of leaving it as a comment here? Also, is this expected to be done before this feature goes to PROD? I'm concerned about leaving a potential infinite sync loop case in PROD,

That comment was mainly for developers to let them know why there is a new sync config. Added the CH story for removing the sync config [here](https://app.clubhouse.io/simpledotorg/story/4125/remove-drugs-sync-config-after-batching-support-is-added-to-medications-api)

Reply:
Hmm, one thing we should do then before pushing this to `PROD` is to decouple updating the remote config from the sync resources. This is something I've been meaning to do for a while because there is no reason for remote config to be synced in the same way.

This will allow us to change the batch size to support older versions since the remote config will get updated regardless of whether or not the drug sync gets into one. What do you think?

Reply:
Yep, that sounds good. We can move this into draft until that PR is merged in if you want.

Reply:
Not necessary, we can merge this in and then pick that up on priority.

Reply:
Would you be able to take a crack at this? We essentially need to add a new work manager job to keep the remote config updated.

Reply:
Okay, will do that 👍🏾